ui/packages/catalog-ui-search/src/main/webapp/component/query-settings/query-settings.view.js

```diff
@@ -55,26 +53,50 @@ define([
         initialize: function () {
             this.model = this.model._cloneOf ? store.getQueryById(this.model._cloneOf) : this.model;
             this.listenTo(this.model, 'change:sortField change:sortOrder change:src change:federation', Common.safeCallback(this.onBeforeShow));
+            if(properties.hasExperimentalEnabled())
+            {
+                this.resultFormCollection = ResultForm.getResultCollection();
+                this.listenTo(this.resultFormCollection, 'change:added', this.handleFormUpdate)
+            }
+        },
+        handleFormUpdate: function(newForm) {
+            this.renderResultForms(this.resultFormCollection.filteredList)
         },
         onBeforeShow: function () {
             this.setupSortFieldDropdown();
+            this.setupSrcDropdown();
             this.turnOnEditing();
-
-            let resultTemplates = ResultForm.getResultTemplatesProperties();
+            if(properties.hasExperimentalEnabled())
+            {
+                this.renderResultForms(this.resultFormCollection.filteredList)
+            }
+        },
+        renderResultForms: function(resultTemplates){
+            if(resultTemplates == undefined)
+            {
+                resultTemplates = [];
+            }
+            resultTemplates.push({
+                label: 'All Fields',
+                value: 'All Fields',
+                id: 'All Fields',
+                descriptors: [],
+                description: 'All Fields'
+            });
+            resultTemplates =  _.uniq(resultTemplates, 'id');
             let lastIndex = resultTemplates.length - 1;
-            if (resultTemplates) {
                 let detailLevelProperty = new Property({
-                    label: 'Detail Level',
-                    enum: ResultForm.getResultTemplatesProperties(),
-                    value: [this.model.get('detail-level') || (resultTemplates && resultTemplates[lastIndex] && resultTemplates[lastIndex].value)],
-                    id: 'Detail Level'
+                label: 'Result Form',
```

Comment:
❓ I thought the verbiage we had decided on was `Detail Level`? Are we sure we want to be changing this to `Result Form`?

Reply:
This was the customer's feedback, and there is some danger with "detail level" ... it implies something like full, summary, brief... as opposed to categorical, like imagery, report, sensor, etc.

Reply:
Detail-Level is confusing and vague; i recommend using `Result Form`